projects/fishtornado_xr/FishTornado.h

```diff
@@ -122,6 +125,21 @@ class FishTornadoApp
         ppx::grfx::FencePtr         uiRenderCompleteFence;
     };
 
+    struct MetricsData
+    {
+        ppx::metrics::Manager                   manager;
+        std::vector<ppx::metrics::MetricGauge*> allMetrics;
```

Comment:
Is there any reason you use an array AND direct pointers?
Couldn't you have something like the pseudo-code below?
```
    enum { kGpuFrameTimeGauge, kCpuFrameTime, ... };
    ...
    allMetrics[kGpuFrameTimeGauge] = run->AddMetric(...); //assuming reserve(), but append works too
    ...
    allMetrics[kGpuFrameTimeGauge]->RecordEntry(now, prevGpuFrameTime);
```
Then, you would not need to keep the pointers in mMetricsData.

Reply:
Oh. Yeah, 100%. I'm being daft. One sec.